### DISCUSSION TITLE
mini.comment commentstring update

### DISCUSSION THREAD

**Discussion by krishna-bala** [LazyVim/LazyVim#776] [Q&A] [answered]
I'm struggling to update the commentstring for specific filetypes. 

Here's my `nvim-treesitter` plugin customization:
```lua
-- lua/plugins/nvim-treesitter.lua
return {
  "nvim-treesitter/nvim-treesitter",
  dependencies = {
    "JoosepAlviste/nvim-ts-context-commentstring",
  },
  opts = {
    context_commentstring = {
      enable = true,
      enable_autocmd = false,
      config = {
        c = "// %s",
        cpp = "// %s",
      },
    },
  },
}
```

Here's my `mini.comment` plugin customization:

```lua
-- lua/plugins/mini-comment.lua
return {
  "echasnovski/mini.comment",
  dependencies = {
    "nvim-treesitter/nvim-treesitter",
  },
  opts = {
    hooks = {
      pre = function()
        print("Successfully calling pre hook function...")
        require("ts_context_commentstring.internal").update_commentstring()
      end,
    },
  },
}
```

I'm following the instructions on the [nvim-ts-context-commentstring Integrations wiki](https://github.com/JoosepAlviste/nvim-ts-context-commentstring/wiki/Integrations#minicomment). I'm able to print my debug message when commenting, but it still uses the default commentstring `/* %s */`.

Any ideas what might be wrong?

**Comment by folke** [comment]
Since recently, `mini.comment` has treesitter comment strings support out of the box and no longer requires ts-context-commentstring.

I removed that package from LazyVim recently. See https://github.com/LazyVim/LazyVim/releases/tag/v3.0.0

**Reply by krishna-bala** [reply]
Thanks for the quick reply. I saw that but am not sure how to update the commentstring now. I tried searching for how to update the commentstring in treesitter, but all my searches brought me to `nvim-ts-context-commentstring`.

**Reply by folke** [reply]
As I mentioned, it is no longer needed to update the commentstring. **mini.comment** does this automatically.

**Reply by krishna-bala** [reply]
Sorry for not being clear. How would I change the default commentstring from `/* %s */` to `// %s` for `c/cpp` filetypes?

**Reply by folke** [reply]
No idea. Best to check the mini.comment docs.

**Reply by dpetka2001** [reply, marked as answer]
@krishna-bala I believe this [issue](https://github.com/echasnovski/mini.nvim/issues/124) at mini.nvim should help you out. Or you could still install `nvim-ts-context-commentstring` and configure your comments in treesitter spec. Choose whatever approach fits you best.

**Reply by krishna-bala** [reply]
Thank you! Yes that solves the problem. I created a [discussion on the mini.nvim repo](https://github.com/echasnovski/mini.nvim/discussions/332) as well (just now seeing your message). For quick reference, the solution that worked for me was:

Create file `after/ftplugin/<some-filetype>.lua` inside your Neovim config (e.g. `~/.config/nvim/`) and put this line there: `vim.bo.commentstring = '<some-commentstring>'`
